sa-token-core/src/main/java/cn/dev33/satoken/util/ExcelUtils.java

```diff
@@ -0,0 +1,215 @@
+		WritableSheet sheet = wtwb.createSheet(title, 0);
+		sheet.getSettings().setDefaultColumnWidth(12);
+
+		// 标题的格式-红色
+		WritableFont redWF = new WritableFont(WritableFont.ARIAL, 12,
+				WritableFont.BOLD, false, UnderlineStyle.NO_UNDERLINE, Colour.RED);
+		WritableCellFormat redWFFC = new WritableCellFormat(redWF);
+		redWFFC.setVerticalAlignment(VerticalAlignment.CENTRE);
+		redWFFC.setBorder(jxl.format.Border.ALL,jxl.format.BorderLineStyle.THIN);
+
+		// 标题的格式-黑色
+		WritableFont blackWF = new WritableFont(WritableFont.ARIAL, 12,
+				WritableFont.BOLD, false, UnderlineStyle.NO_UNDERLINE, Colour.BLACK);
+		WritableCellFormat blackWFFC = new WritableCellFormat(blackWF);
+		blackWFFC.setVerticalAlignment(VerticalAlignment.CENTRE);
+		blackWFFC.setBorder(jxl.format.Border.ALL,jxl.format.BorderLineStyle.THIN);
+
+		// 设置字体以及单元格格式
+		WritableFont wfont = new WritableFont(WritableFont.createFont("楷书"), 12);
+		WritableCellFormat format = new WritableCellFormat(wfont);
+		format.setAlignment(Alignment.LEFT);
+		format.setVerticalAlignment(VerticalAlignment.TOP);
+		format.setBorder(jxl.format.Border.ALL,jxl.format.BorderLineStyle.THIN);
+
+		// 第一行写入提示
+		if(StringUtil.isNotEmpty(tip) && tip.contains("*")) {
+			sheet.addCell(new Label(0, 0, tip, redWFFC));
+		} else {
+			sheet.addCell(new Label(0, 0, tip, blackWFFC));
+		}
+
+		// 第二行写入标题
+		for (int i = 0; i < names.length; i++) {
+			if(StringUtil.isNotEmpty(names[i]) && names[i].contains("*")) {
+				sheet.addCell(new Label(i, 1, names[i], redWFFC));
+			} else {
+				sheet.addCell(new Label(i, 1, names[i], blackWFFC));
+			}
+		}
+
+		// 其余行依次写入数据
+		int rowNum = 2;
+		for (int j = 0; j < objects.size(); j++) {
+			String[] obj = objects.get(j);
+			for (int h = 0; h < obj.length; h++) {
+				sheet.addCell(new Label(h, rowNum, obj[h], format));
+			}
+			rowNum = rowNum + 1;
+		}
+		wtwb.write();
+		wtwb.close();
+		return excelFile;
+	}
+
+	public static String getContent(Sheet src, int rowNum, int colNum) {
+		if(colNum < src.getRow(rowNum).length) {
+			return src.getRow(rowNum)[colNum].getContents().trim();
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * 获取真实的行数，剔除掉空白行
+	 * @param src
+	 * @return
+	 */
+	public static int getRightRows(Sheet src) {
+		int rsRows = src.getRows(); //行数
+		int rsCols = src.getColumns(); //列数
+		int nullCellNum;
+		int rightRows = rsRows;
+		for (int i = 1; i < rsRows; i++) { //统计行中为空的单元格数
+			nullCellNum = 0;
+			for (int j = 0; j < rsCols; j++) {
+				String val = src.getCell(j, i).getContents().trim();
+				if (StringUtils.isEmpty(val)) {
+					nullCellNum++;
+				}
+			}
+			if (nullCellNum >= rsCols) { //如果nullCellNum大于或等于总的列数
+				rightRows--; //行数减一
+			}
+		}
+		return rightRows;
+	}
+
+	public static void downloadExcel(File excelFile, String fileName, HttpServletResponse response) throws Exception{
+		response.setContentType("application/octet-stream");
+		fileName = new String(fileName.getBytes("gbk"),"ISO8859_1");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/xiejijun/sa-token/pull/1#issuecomment-2304574048>#12e843</a><span> - 08/22/2024, 12:43 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Use of legacy character encoding name "gbk" instead of standardized charset name
</div>
<div id="fix">
<b>Fix:</b> Replace the legacy encoding name "gbk" with the standardized charset name "GBK" or preferably use UTF-8 for better international support.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -201,1 +201,1 @@
 -		fileName = new String(fileName.getBytes("gbk"),"ISO8859_1");
 +		fileName = new String(fileName.getBytes(StandardCharsets.UTF_8),"ISO-8859-1");
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged